### PR TITLE
3571 handle accidental uploads of findings associated workbooks

### PR DESCRIPTION
**Pull request by sambodeme**
## Description
This PR allows users to delete a worksheet after they have uploaded it and realized it is not needed.
The change only applies to optional worksheets. When users upload an optional worksheet, a new link appears near the edit link to allow users to delete the worksheet.
Upon clicking on the worksheet removal link, a new page is displayed, warning the user about the deletion action and asking for confirmation before proceeding. If confirmed, the worksheet will be deleted and the user will be redirected to the submission progress page. In case of an error, the application stays on the deletion page and displays a user-friendly error message to indicate the issue.
See #3571 for context.

## To Test:
- Upload an optional worksheet. Note that the removal link appears.
- Click on the link and delete the worksheet.
- Confirm the deletion when prompted.
- Verify that you are redirected to the submission progress page.
- Ensure that the worksheet is correctly removed.

Noticed new links from the submission checklist page 
image

And from the upload page

image

## PR checklist: submitters

- [ ]   Link to an issue if possible. If there’s no issue, describe what your branch does. Even if there is an issue, a brief description in the PR is still useful.
- [ ]   List any special steps reviewers have to follow to test the PR. For example, adding a local environment variable, creating a local test file, etc.
- [ ]   For extra credit, submit a screen recording like [this one](https://github.com/GSA-TTS/FAC/pull/1821).
- [ ]   Make sure you’ve merged `main` into your branch shortly before creating the PR. (You should also be merging `main` into your branch regularly during development.)
- [ ]   Make sure you’ve accounted for any migrations. When you’re about to create the PR, bring up the application locally and then run `git status | grep migrations`. If there are any results, you probably need to add them to the branch for the PR. Your PR should have only **one** new migration file for each of the component apps, except in rare circumstances; you may need to delete some and re-run `python manage.py makemigrations` to reduce the number to one. (Also, unless in exceptional circumstances, your PR should not delete any migration files.)
- [ ]   Make sure that whatever feature you’re adding has tests that cover the feature. This includes test coverage to make sure that the previous workflow still works, if applicable.
- [ ]   Make sure the full-submission.cy.js [Cypress test](https://github.com/GSA-TTS/FAC/blob/main/docs/testing.md#end-to-end-testing) passes, if applicable.
- [ ]   Do manual testing locally. Our tests are not good enough yet to allow us to skip this step. If that’s not applicable for some reason, check this box.
- [ ]   Verify that no Git surgery was necessary, or, if it was necessary at any point, repeat the testing after it’s finished.
- [ ]   Once a PR is merged, keep an eye on it until it’s deployed to dev, and do enough testing on dev to verify that it deployed successfully, the feature works as expected, and the happy path for the broad feature area (such as submission) still works.

## PR checklist: reviewers

- [ ]   Pull the branch to your local environment and run `make docker-clean; make docker-first-run && docker compose up`; then run `docker compose exec web /bin/bash -c "python manage.py test"`
- [ ]   Manually test out the changes locally, or check this box to verify that it wasn’t applicable in this case.
- [ ]   Check that the PR has appropriate tests. Look out for changes in HTML/JS/JSON Schema logic that may need to be captured in Python tests even though the logic isn’t in Python.
- [ ]   Verify that no Git surgery is necessary at any point (such as during a merge party), or, if it was, repeat the testing after it’s finished.

The larger the PR, the stricter we should be about these points.

### CODE REVIEW

File: backend/report_submission/views.py [hunk 1]
```diff
@@ -453,3 +479,119 @@ def post(self, request, *args, **kwargs):
...
+                return redirect(request.path)
+            path_name = request.path.split("/")[2]
+            section = self.additional_context[path_name]
+
+            try:
+                excel_files = ExcelFile.objects.filter(
+                    sac=sac, form_section=section["form_section"]
+                )
+                logger.info(f"Deleting {excel_files.count()} files.")
+                excel_files.delete()
```
Comment (timoballard) [inline backend/report_submission/views.py:572]: Deleting the `ExcelFile` record does not, it turns out, delete the actual file from S3. The data is correctly nulled out in the `SingleAuditChecklist` record, and the `ExcelFile` record is gone, but the S3 file remains (see minio screenshot after deleting the CAP upload):
image

Comment (sambodeme) [reply backend/report_submission/views.py:572]: Yes, this is what we discussed during dev huddle last week and my understanding was that we are not removing from s3 as part of this PR but we'll have another task/ticket to handle the deletion as part dissemination (and possibly clean up files for the already disseminated records).

Comment (timoballard) [reply backend/report_submission/views.py:572]: Right- couldn't 100% recall where we landed last week so I wanted to be sure - thanks for the clarification!

File: backend/report_submission/test_views.py [hunk 2]
```diff
@@ -758,3 +763,105 @@ def test_post_validates_general_information(self):
...
+
+    def test_get_no_access(self):
+        request = self.make_request("get")
+        with patch("audit.models.SingleAuditChecklist.objects.get") as mock_get:
+            mock_get.side_effect = SingleAuditChecklist.DoesNotExist
+            with self.assertRaises(PermissionDenied):
+                self.view.get(request, report_id=self.report_id)
+
+    def test_get_successful_render(self):
+        sac = MagicMock(report_id=self.report_id)
```
Comment (timoballard) [inline backend/report_submission/test_views.py:813]: what are the tradeoffs here between using `MagicMock` and `Baker` (which we use in most of our test code)?

Comment (sambodeme) [reply backend/report_submission/test_views.py:813]: Baker could be used, but I did not find it necessary here as the focus is more on testing the behavior of the view rendering, so the database interaction was mocked. Using Baker would become relevant if it were necessary to test the database interaction with the Django test database.

**Review by Leighdiddy — COMMENTED** [review]
Reviewing for UI only from screenshots - looks good to me!

**Comment by lauraherring** [comment]
Looks good to me from a content perspective.

**Pull request merged**